### ISSUE TITLE
Compilation errors in GuideReactiveTest.kt

### ISSUE THREAD

**Issue by IgorGanapolsky** [Kotlin/kotlinx.coroutines#663]
Hello,
I imported the project in Intellij.  The build fails because of compilation errors in GuideReactiveTest.kt:
> Type inference failed.  Not enough information to infer parameter T in inline fun

<img width="722" alt="screen shot 2018-10-04 at 9 03 56 am" src="https://user-images.githubusercontent.com/201209/46475774-74a7e100-c7b4-11e8-83b2-ac1e3e11edf8.png">

**Comment by elizarov** [comment]
I cannot reproduce it. Can you, please, give more details on your environment -- what versions of IDEA and Kotlin plugin are you using? How did you import the project into IDEA?

**Comment by IgorGanapolsky** [comment]
I am running IntelliJ IDEA 2018.3 EAP (Community Edition)
Build #IC-183.3283.2, built on October 2, 2018
JRE: 1.8.0_152-release-1343-b3 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.1

I imported the project by clicking **File -> New Project From Existing Sources.**
I think I caused the issue by upgrading to `kotlin_verion 1.3.0-rc-80`.  When I reverted all the repo changes, everything runs now.

Thanks!
